### DISCUSSION TITLE
Work for week 1

### DISCUSSION THREAD

**Discussion by richardreeve** [IBAHCM/RPiR#120] [Archive]
@IBAHCM/rpir - this is just to confirm what we've said in the classes this week. First, you should have updated the `RPiR` package today, and you will need to update it again on Friday to include Friday's new lectures and any corrections to the exercises. You can do this by running:

```R
library(devtools)
install_github("IBAHCM/RPiR")
```

If you are having problems with the installation or with running individual practicals (which may be a hidden installation issue), see the [package webpages](https://ibahcm.github.io/RPiR/articles/pages/install_RPiR.html#potential-problems) in the first instance. You can also ask for help in this discussion forum outside the Zoom meetings [here](https://github.com/IBAHCM/RPiR/discussions/categories/q-a-for-technical-issues) - check first that your question hasn't already been answered of course.

We're expecting you to finish the first series of practical exercises this week. You can run them as follows:

```R
library(RPiR)
run_practical("practical0-1")
# Remember you need to hit the stop button or the escape key in the console to stop a practical running
run_practical("practical1-1")
run_practical("practical1-2")
# and so on for 1-3 and 1-4 ...
run_practical("practical1-5")
# You can skip 1-6, as it's not really working!
```

We're not expecting you to submit anything from these practicals, your marks for at least this first week will come from engagement, such as attendance at lectures, responding to the mentimeter and other quizzes, and engaging with this GitHub site (as you've already done by registering with the organisation).

The final work this week is to watch the keynote and the BES Reproducible Research booklet available from [Moodle](https://moodle.gla.ac.uk/course/view.php?id=23343), and to look at [the style guide](http://adv-r.had.co.nz/Style.html) mentioned in 0-1 (and below).

**Comment by soniamitchell** [comment]
Just to emphasise, please **don't** use `install_github("IBAHCM/RPiR", dependencies = FALSE)`. If you've done this, you might not have the correct dependencies installed and Practical 1-2 will appear broken. To fix this, use `install_github("IBAHCM/RPiR", force = TRUE)` and **select 1** to install all dependencies.

**Comment by richardreeve** [comment]
Note there is now an updated version of `RPiR`, so download and install it using:
```R
devtools::install_github("IBAHCM/RPiR")
```
The main change is that there is now a new introductory practical - "practical0-1" - which introduces loops and functions that we use in 1-1 and 1-2, respectively.

**Reply by richardreeve** [reply]
We also recommend reading [this style guide](http://adv-r.had.co.nz/Style.html) on writing clear code in R. It's written by Hadley Wickham, who is the lead developer for RStudio. Generally, if you want a very detailed reference for advanced programming in R, you could do a lot worse than to look at [the whole site](http://adv-r.had.co.nz), and [the updated second edition](https://adv-r.hadley.nz/index.html). There is an enormous amount of useful information there.

**Reply by richardreeve** [reply]
@IBAHCM/rpir I've now updated the instructions at the top of this discussion to be up to date.
